### ISSUE TITLE
Draggable widget conflicts with child GestureDetector onPanUpdate

### ISSUE THREAD

**Issue by shenwilly** [flutter/flutter#28202]
`onPanUpdate` doesn't seem to work when the `GestureDetector` is a child of a `Draggable`, while `onHorizontalDragUpdate` and `onVerticalDragUpdate` do work. 

Complete runnable reproduction `main.dart`: 
https://gist.github.com/Monoss/3078aa2f853e9b9bad810f5ec35c579c

## Steps to Reproduce

1. Switch between `onPanUpdate` and `onHorizontalDragUpdate` &`onVerticalDragUpdate` .
2. Pan the gesture detector (yellow container) .

## Sample Code

```dart
Draggable(
  child: Container(
    child: Row(
      children: <Widget>[
        Container(
          width: 100,
          height: 100,
          color: Colors.red
        ),
        GestureDetector(
          onPanUpdate: (details) {
            print("panning");
          },
//        onHorizontalDragUpdate: (details) {
//        print("draggin");
//        },
          child: Container(
            width: 100,
            height: 100,
            color: Colors.yellow
          )
        )
      ],
    )
  ),
  childWhenDragging: Container(),
  feedback: Container(
      child: Row(
        children: <Widget>[
          Container(
              width: 100,
              height: 100,
              color: Colors.red
          ),
          Container(
              width: 100,
              height: 100,
              color: Colors.yellow
          )
        ],
      )
  ),
)
```
## Logs

```
[√] Flutter (Channel stable, v1.0.0, on Microsoft Windows [Version 10.0.17134.590], locale en-ID)
    • Flutter version 1.0.0 at C:\src\flutter
    • Framework revision 5391447fae (3 months ago), 2018-11-29 19:41:26 -0800
    • Engine revision 7375a0f414
    • Dart version 2.1.0 (build 2.1.0-dev.9.4 f9ebf21297)

[!] Android toolchain - develop for Android devices (Android SDK 28.0.3)
    • Android SDK at C:\Users\ASUS\AppData\Local\Android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    X Android license status unknown.

[√] Android Studio (version 3.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 32.0.1
    • Dart plugin version 182.5124
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[√] IntelliJ IDEA Community Edition (version 2018.3)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2018.3.4
    • Flutter plugin version 33.0.6
    • Dart plugin version 183.5901

[√] Connected device (1 available)
    • Mi 4c • eaad90e6 • android-arm64 • Android 7.0 (API 24)
```

**Comment by zoechi** [comment]
Can you please make the code example complete so that it is runnable (incl `main() {...}`, imports, ...). Thanks!

**Comment by Monoss** [comment]
> Can you please make the code example complete so that it is runnable (incl `main() {...}`, imports, ...). Thanks!

updated!

**Comment by Frozmen** [comment]
Faced the same problem. 
Any updates?

**Comment by januwA** [comment]
[GestureDetector.onPanUpdate] didn't work under [ListView], can anyone help me solve it? 🥺

```dart
 ListView(
        children: [
          Center(
            child: GestureDetector(
              onPanUpdate: (DragUpdateDetails d) async {
               // no work
              },
              child: Container(
                constraints: BoxConstraints(
                  maxWidth: 200,
                  maxHeight: 300,
                ),
                color: Colors.red,
              ),
            ),
          ),
        ],
      ),
```

**Comment by vikalpnagar** [comment]
Moving GestureDetector above Draggable solved the problem for me. 
`GestureDetector(
        onTap: () => ...,
        child: Draggable(`

**Comment by iapicca** [comment]
Hi @shenwilly 
@vikalpnagar 's solution seems to solve the problem
```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) => MaterialApp(
      home: MyHomePage(),
  );
}

class MyHomePage extends StatefulWidget {

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {

  void _snack(BuildContext context, String message) => Scaffold.of(context)
    .showSnackBar(SnackBar(content: Text(message),),);
 
  @override
  Widget build(BuildContext context)=> Scaffold(
    body: Builder(builder: (context) => GestureDetector(
        onTap: () => _snack(context, 'gesture tapping'),
        child: Draggable(
          child: const FlutterLogo(size: 160), 
          feedback: const FlutterLogo(size: 80), 
        ),
      ),
    ),
  );
}
```
does it work for you?
Thank you

**Comment by iapicca** [comment]
Without additional information we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by mehdizarepour** [comment]
Hi @shenwilly, have you found any alternative solution?

**Comment by thecodepapaya** [comment]
@mehdizarepour Not sure but [this StackOverflow answer](https://stackoverflow.com/a/52910582/10505839) could help. It worked for me with the draggable widget.
TL;DR
Use a listener
```
Listener(
  onPointerMove: (ptr){
    print("Coordinates: ${ptr.position}");
  }
  child: Draggable(...),
)
```

**Comment by lau1944** [comment]
``` dart
GestureDetector(
    onPanUpdate: (detail) {
      print('first');
   }
   child: GestureDetector(
       onPanUpdate: (detail)  {
           print('second');
       }
  )
)
```

As my knowlege of GestureArenaManager, 
The code above would dispatch the child event and the parent gesture would be dropped,
but the actual print result is 

```
   'second'
   'first'
```
This issue only appeared at onPanDown() method as I know, onTap(), onVerticalDragDown(), etc.. don't have this issue.

**Comment by viandmarket25** [comment]
Wrap your draggables widget with a listener


as follows:

 Listener(
            onPointerMove: (moveEvent){
              if(moveEvent.delta.dx > 8) {
                print("swipe right");
                //-----or call your function
              }else if(moveEvent.delta.dx < 0){
                print("swipe left");
               //-----or call your function
              }
            },
          child: AnyWiget(),

)

//-------using 8 as the min horizontal swipe value enables you get accurate left and right swipe update, I tried other values starting from zero, sometimes, you get a result opposite your swipe direction
